permissions-policy/resources/orientation-event.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+async function waitForOrientationEvent(eventName) {
+  if (eventName !== 'devicemotion' && eventName !== 'deviceorientation' && eventName !== 'deviceorientationabsolute') {
+    return 'ERROR';
+  }
+
+  let value;
+
+  try {
+    // See https://github.com/w3c/deviceorientation/issues/148: the
+    // specification currently does not fire any events when the permissions
+    // policy checks fail, so what we do here is wait for the timeout and the
+    // devicemotion event handler to race each other.
+    value = await new Promise((resolve, reject) => {
+      const timeoutId = window.setTimeout(() => {
```

Comment:
This won't fly... it's gonna be super racy on testing infrastructure.

Reply:
Do you have anything else in mind? This API doesn't throw any errors or fire any events when a permissions/permissions policy check fails, so the only other option would be adding a ReportingObserver and waiting for a [permissions-policy-violation report](https://w3c.github.io/webappsec-permissions-policy/#reporting). Would that be acceptable?